### PR TITLE
respect GOPRIVATE and GONOPROXY, and relax error condition on proxy queries

### LINKED ISSUE #4: ugbt list --all returns cryptic error
On most binaries I have tried it, the only result returned by `ugbt list --all <binary>` is this error string:

```ugbt: query proxy: 410 Gone```

The only ones where I don't get this error are open source packages, like https://gitlab.com/fgmarand/gocoverstats, so it might be an issue with private repositories and maybe not taking `$GOPRIVATE` into account.

### PR DESCRIPTION
**Pull request by kortschak**
/cc @fgm

Fixes #4.

### CODE REVIEW

File: cmd.go [hunk 1]
```diff
@@ -655,7 +671,7 @@ func (u *ugbt) stdInfo(ctx context.Context) ([]info, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
```
Comment (fgm) [inline cmd.go]: IMHO the code probably wants to accept anything from 200 to 299, not just 200.

File: cmd.go [hunk 2]
```diff
@@ -683,7 +699,7 @@ func (u *ugbt) info(ctx context.Context, version string) (info, error) {
 		return info{}, err
 	}
 	if resp.StatusCode != http.StatusOK {
```
Comment (fgm) [inline cmd.go]: Same here.

File: cmd.go [hunk 3]
```diff
@@ -708,7 +724,7 @@ func (u *ugbt) retractions(ctx context.Context, version string) ([]*modfile.Retr
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
```
Comment (fgm) [inline cmd.go]: And again

File: cmd.go [hunk 4]
```diff
@@ -620,6 +633,9 @@ func (t *ugbt) availableVersions(ctx context.Context, mod string) ([]info, error
 
 			i, err := t.info(ctx, url)
 			if err != nil {
+				if errors.Is(err, statusError{}) {
```
Comment (fgm) [inline cmd.go]: The doc spec says errors other than 404 and 410 should abort https://go.dev/doc/go1.13#modules . Maybe differentiate handling there ?

**Review by fgm — COMMENTED** [review]
Some error handling questions

File: cmd.go (hunk 1 above)
Comment (kortschak) [reply cmd.go]: The spec mandates 200 OK for a GOPROXY query success.

File: cmd.go (hunk 4 above)
Comment (kortschak) [reply cmd.go]: Done

**Comment by kortschak** [comment]
I just want to check that this is now working for you in a way that is more helpful.

**Pull request merged**